server/handlers_test.go

```diff
@@ -0,0 +1,644 @@
+	assert.Equal(http.StatusOK, resp.StatusCode)
+	assert.Equal("success", strings.TrimSpace(string(body)))
+}
+
+func TestMustHaveFormParams_MultipleParamsRequiredOneNotProvided(t *testing.T) {
+	handler := mustHaveFormParams(dummyHandler(), "a", "b")
+
+	form := url.Values{
+		"a": {"foo"},
+	}
+	resp := httpResp(handler, "POST", strings.NewReader(form.Encode()))
+	body, _ := ioutil.ReadAll(resp.Body)
+
+	assert := assert.New(t)
+	assert.Equal(http.StatusBadRequest, resp.StatusCode)
+	assert.Equal("missing form param: b", strings.TrimSpace(string(body)))
+}
+func TestMustHaveFormParams_MultipleParamsRequiredAllProvided(t *testing.T) {
+	handler := mustHaveFormParams(dummyHandler(), "a", "b")
+
+	form := url.Values{
+		"a": {"foo"},
+		"b": {"foo"},
+	}
+	resp := httpResp(handler, "POST", strings.NewReader(form.Encode()))
+	body, _ := ioutil.ReadAll(resp.Body)
+
+	assert := assert.New(t)
+	assert.Equal(http.StatusOK, resp.StatusCode)
+	assert.Equal("success", strings.TrimSpace(string(body)))
+}
+
+func TestFundAndApproveSignersHandler_MissingClient(t *testing.T) {
+	handler := fundAndApproveSignersHandler(nil)
+
+	resp := httpResp(handler, "POST", nil)
+	body, _ := ioutil.ReadAll(resp.Body)
+
+	assert := assert.New(t)
+	assert.Equal(http.StatusInternalServerError, resp.StatusCode)
+	assert.Equal("missing ETH client", strings.TrimSpace(string(body)))
+}
+
+func TestFundAndApproveSignersHandler_InvalidAmount(t *testing.T) {
+	client := &eth.MockClient{}
+	handler := fundAndApproveSignersHandler(client)
+
+	form := url.Values{
+		"amount": {"foo"},
+	}
+	resp := httpResp(handler, "POST", strings.NewReader(form.Encode()))
+	body, _ := ioutil.ReadAll(resp.Body)
+
+	assert := assert.New(t)
+	assert.Equal(http.StatusBadRequest, resp.StatusCode)
+	assert.Equal("invalid amount", strings.TrimSpace(string(body)))
+}
+
+func TestFundAndApproveSignersHandler_MinPenaltyEscrowError(t *testing.T) {
+	client := &eth.MockClient{}
+	handler := fundAndApproveSignersHandler(client)
+
+	client.On("MinPenaltyEscrow").Return(nil, errors.New("MinPenaltyEscrow error"))
+
+	form := url.Values{
+		"amount": {"100"},
+	}
+	resp := httpResp(handler, "POST", strings.NewReader(form.Encode()))
+	body, _ := ioutil.ReadAll(resp.Body)
+
+	assert := assert.New(t)
+	assert.Equal(http.StatusInternalServerError, resp.StatusCode)
+	assert.Equal("could not execute fundAndApproveSigners", strings.TrimSpace(string(body)))
+}
+
+func TestFundAndApproveSignersHandler_InsufficientAmount(t *testing.T) {
+	client := &eth.MockClient{}
+	handler := fundAndApproveSignersHandler(client)
+
+	client.On("MinPenaltyEscrow").Return(big.NewInt(101), nil)
+
+	form := url.Values{
+		"amount": {"100"},
+	}
+	resp := httpResp(handler, "POST", strings.NewReader(form.Encode()))
+	body, _ := ioutil.ReadAll(resp.Body)
+
+	assert := assert.New(t)
+	assert.Equal(http.StatusBadRequest, resp.StatusCode)
+	assert.Equal("amount is not sufficient for minimum penalty escrow", strings.TrimSpace(string(body)))
```

Comment:
Perhaps it's valuable to include the minimum amount from the contract in this response (for better UX)?

Reply:
In the latest updates, the server no longer checks if the amount is sufficient for the minimum penalty escrow - the wizard client does these checks instead when prompting the user for an amount to deposit